### PR TITLE
Implement onDisplayPlatformView

### PR DESCRIPTION
**Pull request by blasten**
## Description

Implements onDisplayPlatformView

## Related Issues

Fixes https://github.com/flutter/flutter/issues/60140

## Tests

I added the following tests: Scenario test app

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformViewsChannel.java [hunk 1]
```diff
@@ -216,18 +234,29 @@ public void setPlatformViewsHandler(@Nullable PlatformViewsHandler handler) {
...
+     * <p>The Android {@code View} is added to the view hierarchy.
      */
-    // TODO(mattcarroll): Introduce an annotation for @TextureId
-    long createPlatformView(@NonNull PlatformViewCreationRequest request);
+    void createAndroidViewForPlatformView(@NonNull PlatformViewCreationRequest request);
 
     /**
-     * The Flutter application could like dispose of an existing Android {@code View}, i.e.,
-     * platform view.
+     * The Flutter application would like dispose of an existing Android {@code View} rendered in
```
Comment (jason-simmons) [inline shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformViewsChannel.java]: typo: "would like to dispose"

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java [hunk 2]
```diff
@@ -94,12 +93,34 @@
...
+            ((FlutterView) flutterView).addView(platformViews.get(viewId));
+            platformViews.remove(viewId);
+          }
+        }
 
+        @Override
+        public long createVirtualDisplayForPlatformView(
+            @NonNull PlatformViewsChannel.PlatformViewCreationRequest request) {
+          // API level 20 is required for VirtualDisplay#setSurface which we use when resizing a
+          // platform
```
Comment (jason-simmons) [inline shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java:121]: typo: put "platform view" on one line

File: shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformViewsChannel.java (hunk 1 above)
Comment (blasten) [reply shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformViewsChannel.java]: Done.

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java (hunk 2 above)
Comment (blasten) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java]: Done.

**Comment by blasten** [comment]
Merging. "Mac iOS Engine" failure is already fixed on master.

**Comment by xster** [comment]
go/tested-engine-prs nag :)
Please add tests to PRs with code

**Comment by blasten** [comment]
@xster these changes are tested via the scenario test app and e2e driver tests. Unit tests were out of scope in this change.

**Pull request merged**